### ISSUE TITLE
immutablejs combinereducers example has high overhead

### ISSUE THREAD

**Issue by bdwain** [redux-loop/redux-loop#51]
When used with immutablejs as described in the readme, combineReducers looks like it might suffer from high overhead because each time you call the setState method, immutablejs will create a new copy of the store. It would be more efficient if the mutations were batched with the immutablejs method withMutations. (basically the provided combinereducers but with the main block wrapped by "withMutations"

I wouldn't expect a method optimized for immutablejs to be provided by this repo, but as long as it's being mentioned in the readme, it might be worthing pointing out that performance caveat so people know they should probably be writing their own. 

I think something like this would work well

```javascript
function immutableJsCombineReducers(reducerObj) {
  const reducerKeys = Object.keys(reducerObj);

  return (rootState = Map(), action) => {
    let hasChanged = false;
    let effects = [];

    const model = rootState.withMutations(initialState => {
      reducerKeys.reduce((previousState, reducerName) => {
        const currentChildState = previousState.get(reducerName);
        let nextChildState = reducerObj[reducerName](currentChildState, action);

        if (isLoop(nextChildState)) {
          effects.push(getEffect(nextChildState));
          nextChildState = getModel(nextChildState);
        }

        hasChanged = hasChanged || nextChildState !== currentChildState;
        return previousState.set(reducerName, nextChildState);
      }, initialState);
    });

    return loop(
      hasChanged ? model : state,
      optimizeBatch(effects)
    );
  };
}
```

**Comment by bdwain** [comment]
I just realized that the package doesn't export isLoop, so it could be hard to write this. i'll make a PR to export it

**Comment by bdwain** [comment]
closed since #53 was merged
